### PR TITLE
Add test case, Class based view and jamin's feedback points.

### PR DESCRIPTION
**Pull request by arpanpersistent**
Implementation Details:

1. Convert function based view into class based view.
2. Add test case code for all validation and functionality.
3. Removed onclick in template and use <a> tag.

Testing Details:
1. after converting into class based view , everything in working fine.
2. Add test case and check all cases are working fine.

### CODE REVIEW

**Comment by sahilgupta71754** [comment]
Please resolve the merge conflict

File: UI/interface/user/tests/test_add_customer.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        form_data = {'customer_name': 'Test Customer', 'category_name': 'Test Category'}
+        form = CustomerForm(data=form_data)
+        self.assertTrue(form.is_valid())
+
+    # customer and category length not more then 140 character
+    def test_form_validations_with_long_names(self):
+        mock_request = Mock()
+        long_name='a'*141
+        form_data = {'customer_name': long_name, 'category_name': long_name}
+        form = CustomerForm(data=form_data,request=mock_request)
```
[the remark this answers is not in the document]
Comment (arpanpersistent) [reply UI/interface/user/tests/test_add_customer.py]: done

File: UI/interface/user/tests/test_add_customer.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        self.client.login(username='testuser', password='12345')
+        response = self.client.get(reverse('add_customer'))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'user/customer.html')
+
+    # customer and category not allowed any special characters
+    def test_form_validations_with_special_characters(self):
+        mock_request = Mock()
+        form_data = {'customer_name': 'Test@Customer', 'category_name': 'Test@Category'}
+        form = CustomerForm(data=form_data,request=mock_request)
```
[the remark this answers is not in the document]
Comment (arpanpersistent) [reply UI/interface/user/tests/test_add_customer.py]: done

**Review by abhishek-p8 — APPROVED** [review]
Approved.

**Review by sahilgupta71754 — APPROVED** [review]
Approved

**Pull request merged**